libvast/builtins/loaders/stdin_loader.cpp

```diff
@@ -0,0 +1,102 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/concept/parseable/to.hpp>
+#include <vast/concept/parseable/vast/data.hpp>
+#include <vast/detail/fdinbuf.hpp>
+#include <vast/detail/string.hpp>
+#include <vast/plugin.hpp>
+
+#include <unistd.h>
+
+namespace vast::plugins::stdin_loader {
+
+// -- loader plugin -----------------------------------------------------
+class plugin : public virtual loader_plugin {
+public:
+  static constexpr inline auto max_chunk_size = size_t{16384};
+
+  [[nodiscard]] auto make_loader(const record&, operator_control_plane&) const
+    -> caf::expected<loader> override {
+    return [this]() -> generator<chunk_ptr> {
+      auto in_buf = detail::fdinbuf(STDIN_FILENO, max_chunk_size);
+      in_buf.read_timeout() = read_timeout;
+      auto current_data = std::vector<std::byte>{};
+      current_data.reserve(max_chunk_size);
+      auto eof_reached = false;
+      while (not eof_reached) {
+        auto current_char = in_buf.sbumpc();
+        if (current_char != detail::fdinbuf::traits_type::eof()) {
+          current_data.emplace_back(static_cast<std::byte>(current_char));
+        } else {
+          eof_reached = (not in_buf.timed_out());
```

Comment:
Can you write an issue to track that this behavior is currently untested? I don't expect a test for it within this PR, as that's much easier to add as an integration test later on.

Reply:
Done.